core-bundle/src/Search/Document.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Search;
+
+use Psr\Http\Message\UriInterface;
+use Symfony\Component\DomCrawler\Crawler;
+
+class Document
+{
+    /**
+     * @var UriInterface
+     */
+    private $uri;
+
+    /**
+     * @var int
+     */
+    private $statusCode;
+
+    /**
+     * An array of headers.
+     * The key is the header name in lowercase letters
+     * and the value is again an array of header
+     * values.
+     *
+     * @var array<string,array>
+     */
+    private $headers = [];
+
+    /**
+     * @var string
+     */
+    private $body;
+
+    /**
+     * @var array|null
+     */
+    private $jsonLds;
+
+    public function __construct(UriInterface $uri, int $statusCode, array $headers = [], string $body = '')
+    {
+        $this->uri = $uri;
+        $this->statusCode = $statusCode;
+        $this->headers = array_change_key_case($headers);
+        $this->body = $body;
+    }
+
```

Comment:
Could a factory method be useful here?

```php
public static function createFromRequestResponse(Request $request, Response $reponse): self
```

Reply:
I like that idea. Added in ef3cb8c08608dd0b370f76b60fdd1ce692db04c3.